### PR TITLE
Forgot password API

### PR DESCRIPTION
**Pull request by ShrijanL**
- Implemented Forgot Password API with integration to the New Password API.
- Integrated Captcha service for Forgot Password and Sign-Up workflows.
- Developed and validated test cases for Forgot Password, New Password, and Captcha service APIs.

### CODE REVIEW

**Review by nareshelequin — CHANGES_REQUESTED** [review]

File: django_generic_api/tests/test_captcha.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import pytest
+from fixtures.API import api_client
+from django.core.cache import cache as cache1
+
+
+@pytest.mark.django_db
+class TestCaptchaAPI:
+
+    def test_captcha_post_success(self, api_client):
```
[the remark this answers is not in the document]
Comment (ShrijanL) [reply django_generic_api/tests/test_captcha.py:9]: added 2 negative testcases.

File: django_generic_api/apps.py [hunk 2]
```diff
@@ -33,3 +33,17 @@ def ready(self):
         for key, value in DEFAULT_DRF_THROTTLE_SETTINGS.items():
             if key not in settings.REST_FRAMEWORK:
                 settings.REST_FRAMEWORK[key] = value
+
+        # Predefining cache settings
+        DEFAULT_CACHE_SETTINGS = {
```
[the remark this answers is not in the document]
Comment (ShrijanL) [reply django_generic_api/apps.py:38]: 3. Updated such that if user defines cache with `default ` as key , it raises error

File: README.md [hunk 3]
```diff
@@ -318,7 +348,9 @@ url: "http://domain-name/api/register/",
         "variables":{
             "email":"user@example.com",
             "password":"123456",
-            "password1":"123456"
+            "password1":"123456",
+            "captcha_id": "<captcha_id>",
+            "captcha_number": <captcha_number>
```
Comment (nareshelequin) [inline README.md:353]: rename to captcha_value.

File: django_generic_api/apps.py [hunk 4]
```diff
@@ -19,17 +17,37 @@ def ready(self):
...
         }
 
         # Apply these defaults directly to `REST_FRAMEWORK` settings
         if not hasattr(settings, "REST_FRAMEWORK"):
-            settings.REST_FRAMEWORK = {}
+            setattr(settings, "REST_FRAMEWORK", {})
 
         # Apply these defaults directly to `api_settings`
         for key, value in DEFAULT_DRF_THROTTLE_SETTINGS.items():
             if key not in settings.REST_FRAMEWORK:
```
Comment (nareshelequin) [inline django_generic_api/apps.py:32]: if key not in settings.REST_FRAMEWORK:
update to `if not settings.REST_FRAMEWORK.get(key):`

File: django_generic_api/apps.py [hunk 5]
```diff
@@ -19,17 +17,37 @@ def ready(self):
...
+        2. If settings does not have cache.default, add cache configuration.
+        """
+        if not hasattr(settings, "CACHES"):
+            setattr(
+                settings,
+                "CACHES",
+                {"default": GENERIC_API_PACKAGE_CACHE_SETTINGS},
+            )
+        else:
+            if "default" not in settings.CACHES:
```
Comment (nareshelequin) [inline django_generic_api/apps.py:52]: update to `if not settings.CACHES.get("default"):`

File: django_generic_api/payload_models.py [hunk 6]
```diff
@@ -74,11 +75,24 @@ def validate_filters(cls, v):
...
+    email: EmailStr
     password: SecretStr
 
 
 class GenericRegisterPayload(BaseModel, PydanticConfigV1):
     email: EmailStr
     password: SecretStr
     password1: SecretStr
+    captcha_id: UUID1
+    captcha_number: int
```
Comment (nareshelequin) [inline django_generic_api/payload_models.py:87]: update to `captcha_value`. and find is there any python captcha default captcha format is there? if yes, user can define? try define with alpha numeric and test.

File: django_generic_api/views.py [hunk 7]
```diff
@@ -405,3 +492,119 @@ def get(self, request, encoded_token, *args, **kwargs):
...
                 status=status.HTTP_400_BAD_REQUEST,
             )
+
+
+class CaptchaServiceAPIView(APIView):
+
+    # post method
+    def post(self, request, *args, **kwargs):
+
+        image_captcha = ImageCaptcha()
```
Comment (nareshelequin) [inline django_generic_api/views.py:502]: Develop HTML and share screenshot? how it's looking with required place?

File: README.md (hunk 3 above)
Comment (ShrijanL) [reply README.md]: done

File: django_generic_api/apps.py (hunk 4 above)
Comment (ShrijanL) [reply django_generic_api/apps.py]: done

File: django_generic_api/views.py (hunk 7 above)
Comment (ShrijanL) [reply django_generic_api/views.py]: captcha_example

File: django_generic_api/apps.py (hunk 5 above)
Comment (ShrijanL) [reply django_generic_api/apps.py]: Removed cache as it is not used

File: django_generic_api/payload_models.py (hunk 6 above)
Comment (ShrijanL) [reply django_generic_api/payload_models.py]: updated with captcha_value

**Review by nareshelequin — APPROVED** [review]

**Pull request merged**